### PR TITLE
cs/feat/bulk check file count

### PR DESCRIPTION
**Pull request by chancesmith**
## overall purpose
What if something changes and we don't get a file count that matches our CSV row count?

## changes made
- add file count checker
- move safety checks to another file
- logged out the zip file location so developer doesn't have to dig to much

## screenshot
Screen Shot 2022-12-30 at 3 38 39 PM

### CODE REVIEW

File: csv-bulk-signature/src/main.ts [hunk 1]
```diff
@@ -4,35 +4,21 @@ import handlebars from "handlebars";
...
   LOGOS,
-  TEMPLATE_FILE,
   SIGNATURES_PATH,
-  ZIP_FILE,
   STATUS_REPORT_FILE,
-  CSV_FILE,
-  CSV_EXPIRATION_DAYS,
+  TEMPLATE_FILE,
+  ZIP_FILE,
 } from "./config";
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:13]: tidy imports

File: csv-bulk-signature/src/main.ts [hunk 2]
```diff
@@ -4,35 +4,21 @@ import handlebars from "handlebars";
...
-
-interface TemplateData {
-  logoUrl: string;
-  fullName: string;
-  credentials: string;
-  title: string;
-  officePhone: string;
-  mobilePhone: string;
-  calendly: string;
-}
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:35]: move types to file

File: csv-bulk-signature/src/main.ts [hunk 3]
```diff
@@ -4,35 +4,21 @@ import handlebars from "handlebars";
...
+import {
+  checkContactsFileCreatedAt,
+  checkFileCountWithCsvCount,
+  checkHeadersToBeSame,
+  checkZipIsNotEmpty,
+  hasContactsFile,
+} from "./safety-checks";
+import { TemplateData, Contact } from "./types";
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:21]: move types and safety-checks to files

File: csv-bulk-signature/src/main.ts [hunk 4]
```diff
@@ -244,9 +170,14 @@ async function main() {
...
+  // 4. generate signatures
   await generateSignatures(template, contactsWithNewProps);
+  await checkFileCountWithCsvCount(contactsWithNewProps);
+
+  // 5. create report
   await createStatusReport(contactsWithNewProps);
+
+  // 6. zip up files
   await zipUpFile();
   await checkZipIsNotEmpty(ZIP_FILE);
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:182]: break up final steps

File: csv-bulk-signature/src/safety-checks.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  const isCountSame = signaturesCount === contactsCount;
+
+  if (!isCountSame) {
+    console.error(
+      `\u001b[31m\u001b[1mError: \u001b[0m\u001b[31mSignatures count does not match contacts count. Please check the contacts.csv file or zipFilePath() and try again.\u001b[0m`
+    );
+  } else {
+    console.log("👍 Signatures count matches CSV count");
+  }
+}
```
Comment (chancesmith) [inline csv-bulk-signature/src/safety-checks.ts:27]: check that generated file count matches CSV row count

**Review by adamnitti — APPROVED** [review]

**Pull request merged**